docs/cli/ilab-local-config.md

```diff
@@ -0,0 +1,184 @@
+
+While this method allows ilab to avoid additional complexity of managing a dedicated directory, it’s limited by its ability to scale beyond constraining the user to run ilab in a single directory.
+Since the new ilab CLI will need to not only manage datasets, downloaded models, and any intermediary files, it'll also have to make it simple for the user to orchestrate these things.
+
+We presently store these paths directly in the config, however this
+isn't sustainable as data is frequently moved around unbeknownst to the program. This leaves the user with two choices: either update
+the config to correctly point to either a relative path in the format
+of `../../uncle-dir/dataset.jsonl`, or worse, have to specify an absolute path.
+
+
+## What's being proposed?
+
+We propose the following:
+
+1. `ilab` will have a dedicated config directory seated at `~/.config/ilab` on Linux/MacOS systems, and the `%APPDATA%` equivalent if we decide to support Windows.
+2. `ilab` will have a dedicated program files directory seated at `~/.local/share/ilab` which will store model checkpoints, downloaded models, etc. 
+3. Expectation of having config migrations in the future as it's evolved on an as-needed basis.
+
+
+## Config Directory
+
+The `ilab` CLI will have a dedicated config directory rooted at
+`~/.config/ilab`.
+
+This will just house the `config.yaml` as-is, and any other
+configuration files that should live here.
+
+
+## Data Directory
+
+`ilab` needs to generate, store, and manage different pieces of data over the course of its lifecycle. This includes model checkpoints, generated data, and various logs from processes such as evaluation.
+
+Files like the model bases are large and take a significant amount of time to download. Similarly, the generated datasets are expensive to re-generate and should be easier to manage.
+
+For these reasons, we propose that this data will have a permanent home at `~/.local/share/ilab`. 
+
+This is a rough outline of all of the data pieces consumed and generated by different points in the process:
+
+| phase | consumed data | generated data |
+|-------|---------------|----------------|
+| SDG | QnA files from taxonomy, teacher model checkpoint | knowledge dataset for training, skills dataset for training |
+| Training | Input dataset, base model | Model checkpoints, processed dataset to be consumed during training |
+| Eval | Model checkpoint being evaluated | JSON data / logs containing the score |
+| Publish | Model checkpoints to be published | n/a |
+
+
+Based on this, it's evident that we need a few different subdirectories:
+
+| path | description |
+|------|-------------|
+| `base_models/` | Stores the base models that we download |
+| `checkpoints/` | Houses the checkpoints we generate during training. Can potentially split these out further into `checkpoints/phase00`, `checkpoints/phase05`, and `checkpoints/phase10` |
```

Comment:
Why would we do this splitting? Which approach is the design doc actually proposing?

Reply:
Because we produce many checkpoints so it may become cumbersome to view all of them vs. for a specific phase.

Reply:
So which approach are you actually proposing here? If it's the former can you add this explanation as to why we might want the latter in the future?

Reply:
Either can be done here. I was hoping to leave this portion open for comments to hear the case for both. But with how we want ilab to work, leaving it as `knowledge` vs. `skills` will probably be fine. If we want to support `phase00` in the CLI then we can always revisit this.

Reply:
Ah I hear ya - honestly I trust your expertise here! Fine with either approach.